cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go

```diff
@@ -119,6 +119,58 @@ func (t *replicateDiskTask) GetResponse() proto.Message {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+func (t *replicateDiskTask) getProxyOverlayDiskID(
+	diskID string,
+	checkpointID string,
+) string {
+
+	return fmt.Sprintf("proxy_%v_%v", diskID, checkpointID)
+}
+
+func (t *replicateDiskTask) createProxyOverlayDisk(
+	ctx context.Context,
+	execCtx tasks.ExecutionContext,
+	client nbs_client.Client,
+	currentCheckpointID string,
+) (string, error) {
+
+	diskID := t.request.SrcDisk.DiskId
+	proxyOverlayDiskID := t.getProxyOverlayDiskID(diskID, currentCheckpointID)
+
+	created, err := client.CreateProxyOverlayDisk(
+		ctx,
+		proxyOverlayDiskID,
+		diskID, // baseDiskID
+		currentCheckpointID,
+	)
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
Don't you need to save the state? What about if the task gets terminated by context?

Reply:
id does not metter

CreateVolume transaction is idempotent in nbs